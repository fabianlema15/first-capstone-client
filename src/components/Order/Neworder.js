import React from 'react';

class NewOrder extends React.Component{
  render(){
    return <main role="main">
      <header role="banner">
        <h1>Dream Walker</h1>
        <h2>just imagine.</h2>
      </header>
      <section>
        <header>
            <h3>New Order</h3>
            <fieldset>
              <label forhtml="client">Client</label>
              <select id="client">
                <option>Mark</option>
              </select>
            </fieldset>
        </header>
        <fieldset>
          <label forhtml="product">Product</label>
          <select id="product">
            <option>Product 1</option>
          </select>
          <label forhtml="quantity">Quantity</label>
          <input type="number" id="quantity"/>
        </fieldset>
        <button type='button'>Add Product</button>
      </section>
      <section>
        <header>
            <h3>List of product in the order</h3>
        </header>
      </section>
      <section>
        <header>
            <h3>Sum total of values</h3>
        </header>
      </section>
    </main>
  }
}

export default NewOrder;
