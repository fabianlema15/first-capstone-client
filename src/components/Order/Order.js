import React from 'react';
import GenericList from '../Generic/GenericList';
import GenericContext from '../../contexts/GenericContext'
import { Link } from 'react-router-dom'
import Header from '../Header/Header'

class Order extends React.Component{
  static contextType = GenericContext;

  componentDidMount() {
    this.context.clearError()
    this.context.getAll()
  }

  render(){
    return <div>
      <Header />
    <main className='content-with-nav'>
        <section>
          <Link to='/orders/new'>Create New Order</Link>
        </section>
        <section>
          <GenericList/>
        </section>
      </main>
    </div>
  }
}

export default Order;
