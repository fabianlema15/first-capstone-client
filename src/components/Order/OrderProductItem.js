import React from 'react';
import GenericContext from '../../contexts/GenericContext'
import Form from '../Utils/Form'
import { ColoredLine } from '../Utils/FormElements'

class OrderProductItem extends React.Component{
  static contextType = GenericContext;

  cancelSubmit = (id) => {
    this.context.setSelected(id, 1, false);
  }

  createText = (key) => {
    let text = this.props.obj[key];
    if (key==='product_id')
      text = this.props.obj.product.name
    return text;
  }

  getItemContent = () => {

    return Object.keys(this.props.objLabel).map((key, idx) => <div key={idx} className="item-description">
          <div><span>{this.props.objLabel[key].label}:</span></div> <div>{this.createText(key)}</div>
       </div>)
  }

  render(){
    return <div>
      {this.props.obj.selected ? <Form type={`Edit ${this.context.objName}`} elements={this.context.getObjArrayProd} obj={this.props.obj} actionSubmit={this.context.submitProductEdit} actionCancel={this.cancelSubmit} /> :
        <div>{this.getItemContent()}
        <button className='red' onClick={e => this.context.delete(this.props.obj.id, 1)}>Delete</button>
        </div>
      }
      <ColoredLine color="gray" />
    </div>
  }
}

export default OrderProductItem;
