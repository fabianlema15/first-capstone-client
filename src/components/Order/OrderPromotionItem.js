import React from 'react';
import GenericContext from '../../contexts/GenericContext'
import Form from '../Utils/Form'
import { ColoredLine } from '../Utils/FormElements'

class OrderPromotionItem extends React.Component{
  static contextType = GenericContext;

  cancelSubmit = (id) => {
    this.context.setSelected(id, 2, false);
  }

  createText = (key) => {
    let text = this.props.obj[key];
    if (key==='promotion_id')
      text = this.props.obj.promotion.name
    return text;
  }

  getItemContent = () => {
    return Object.keys(this.props.objLabel).map((key, idx) => <div key={idx} className="item-description">
          <div><span>{this.props.objLabel[key].label}:</span></div> <div>{this.createText(key)}</div>
       </div>)
  }

  render(){
    return <div className="item-list">
      {this.props.obj.selected ? <Form type={`Edit ${this.context.objName}`} elements={this.context.getObjArrayProm} obj={this.props.obj} actionSubmit={this.context.submitPromotionEdit} actionCancel={this.cancelSubmit} /> :
        <div>{this.getItemContent()}
        <button className='red' onClick={e => this.context.delete(this.props.obj.id, 2)}>Delete</button>
        </div>
      }
      <ColoredLine color="gray" />
    </div>
  }
}

export default OrderPromotionItem;
