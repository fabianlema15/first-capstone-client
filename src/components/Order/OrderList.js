import React from 'react';
import GenericContext from '../../contexts/GenericContext'
import OrderProductItem from './OrderProductItem';
import OrderPromotionItem from './OrderPromotionItem';
import Helper from '../Utils/Helper'

class OrderList extends React.Component{
  static contextType = GenericContext;

  renderObjs() {
    let objList = [];
    if (!this.props.objList){
      objList = this.context.objList
      return Object.keys(objList).map(key =>
        <OrderProductItem key={objList[key].id} obj={objList[key]} objLabel={Helper.serializeObj(this.context.getObjArrayProd)}/>
      )
    }else{
      objList = this.props.objList
      return Object.keys(objList).map(key =>
        <OrderPromotionItem key={objList[key].id} obj={objList[key]} objLabel={Helper.serializeObj(this.context.getObjArrayProm)}/>
      )
    }

  }

  render(){
    return <div>
            {this.renderObjs()}
    </div>
  }
}

export default OrderList;
