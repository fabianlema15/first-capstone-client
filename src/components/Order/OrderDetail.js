import React from 'react';
import GenericNew from '../Generic/GenericNew';
import OrderList from '../Order/OrderList';
import GenericContext from '../../contexts/GenericContext'
import { Textarea, Select } from '../Utils/FormElements';
import Header from '../Header/Header'

class OrderDetail extends React.Component{
  static contextType = GenericContext;

  state = {
    subtotal: 0,
    tax: 0,
    total: 0,
    observation: '',
    client_id: ''
  }

  componentDidMount() {
    this.context.clearError()
    this.context.getAll()
  }

  handleInputChange = e => {
    const input = e.target.id;
    const value = e.target.value;
    this.setState({
      [input]: value
    })
  }

  submitNewOrder = (e) => {
    const newOrderDetail = {
      client_id: this.state.client_id,
      subtotal: this.state.subtotal,
      tax: this.state.tax,
      total: this.state.total,
      observation: this.state.observation,
    }
    this.context.submitNewOrder(newOrderDetail)
  }

  render(){
    return <div>
      <Header back='/orders' />
    <main className='content-with-nav'>
        <section>
          <Select id='client_id' value={this.context.currentOrder?this.context.currentOrder.client_id:this.state.client_id} label='Client' onChange={this.handleInputChange} options={this.context.getClientsObj}/>
          <Textarea id='observation' value={this.state.observation} label='Observation' onChange={this.handleInputChange}/>
          {!this.context.currentOrder && <button className='blue' onClick={this.submitNewOrder}>Create and Save</button>}
          {(!this.context.showForm && this.context.currentOrder) && <div>
              <button className='green' onClick={e => this.context.showHideForm(!this.context.showForm, 1)}>Add Product</button>
              <button className='green' onClick={e => this.context.showHideForm(!this.context.showForm, 2)}>Add Promotion</button>
            </div>}
        </section>
        {this.context.error && <section><div className='errorMsg'>{this.context.error}</div></section>}
        {this.context.showForm && <section><GenericNew /></section>}
        <section>
          {(Object.keys(this.context.objList).length > 0 || Object.keys(this.context.objListAux).length > 0) ? <div>
            <OrderList/>
          <OrderList objList={this.context.objListAux}/></div> : 'You still have no products added'}

        </section>
        <section>
          <div className="item-description">
              <div><span>Subtotal:</span></div><div>{this.context.currentOrder?this.context.currentOrder.subtotal:'0'}</div>
              <div><span>Tax:</span></div><div>{this.context.currentOrder?this.context.currentOrder.tax:'0'}</div>
              <div><span>Total:</span></div><div>{this.context.currentOrder?this.context.currentOrder.total:'0'}</div>
           </div>
        </section>
      </main>
    </div>
  }
}

export default OrderDetail;
