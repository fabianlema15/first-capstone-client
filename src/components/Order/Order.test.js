import Order from './Order'
import { BrowserRouter } from 'react-router-dom'
import React from 'react';
import ReactDOM from 'react-dom';
import toJson from 'enzyme-to-json'
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('<Order />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
      <Order>
      </Order>
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders Order as expected', () => {
    const clearError = () => {}
    const getAll = () => {}
    const context = { objName: 'Test', objList: {}, clearError, getAll};
	  const wrapper = shallow(<BrowserRouter><Order/></BrowserRouter>, { context });
	  expect(toJson(wrapper)).toMatchSnapshot();
	  });

})
