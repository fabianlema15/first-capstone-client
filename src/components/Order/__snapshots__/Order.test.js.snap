// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Order /> Renders Order as expected 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Order />
</Router>
`;
